app/src/main/java/org/mozilla/tv/firefox/pocket/PocketRepoCache.kt

```diff
@@ -24,23 +23,40 @@ import android.arch.lifecycle.MutableLiveData
  */
 open class PocketRepoCache(private val repo: PocketVideoRepo) {
 
-    private val _feedState = MutableLiveData<PocketVideoRepo.FeedState>() // Mutable backer for feedState
-    open val feedState: LiveData<PocketVideoRepo.FeedState> = _feedState
-
     // This should only be unfrozen when Pocket videos are not visible to the user.
     // See class kdoc
     var frozen = false
 
-    fun setup() {
-        repo.feedState.observeForever {
-            // Freezing the cache should prevent new videos from overwriting old ones, but we don't want to (for
-            // example) prevent new videos from overwriting a loading state. cachedValueIsBad means that the cached
-            // value is not a successful load
-            val cachedValueIsBad = _feedState.value !is PocketVideoRepo.FeedState.LoadComplete
-
-            if (!frozen || cachedValueIsBad) {
-                _feedState.postValue(it)
-            }
+    open val feedState : Observable<PocketVideoRepo.FeedState> =
+        repo.feedState.scan { cachedValue: PocketVideoRepo.FeedState, currentValue: PocketVideoRepo.FeedState ->
```

Comment:
great use of `scan`!

another clarity note that we find useful in Lockbox is to use `\n` to help with readability when applying operators. so this would read:
```kotlin
repo.feedState // <- get your observable
    .scan { // <- each operator gets its own `\n`
        ...
    }
    .distinctUntilChanged()
    ...
```